app/controllers/DeductionsSummaryController.scala

```diff
@@ -20,59 +20,82 @@ import audit.AuditingService
+        auditingService.extendedAudit(AllowanceAndDeductionsResponseAuditModel(user, viewModel))
+        Ok(deductionBreakdownView(viewModel, taxYear, backUrl, btaNavPartial = user.btaNavPartial, isAgent = isAgent)(implicitly, messages))
+      case error: LiabilityCalculationError if error.status == NOT_FOUND =>
+        Logger("application").info(s"${if (isAgent) "[Agent]"}[DeductionsSummaryController][showDeductionsSummary[$taxYear]] No deductions data found.")
+        itvcErrorHandler.showInternalServerError()
+      case _: LiabilityCalculationError =>
+        Logger("application").error(
+          s"${if (isAgent) "[Agent]"}[DeductionsSummaryController][showDeductionsSummary[$taxYear]] No new calc deductions data error found. Downstream error")
+        itvcErrorHandler.showInternalServerError()
+    }
+  }
 
-    action.async {
+  def showDeductionsSummary(taxYear: Int): Action[AnyContent] =
+    (checkSessionTimeout andThen authenticate andThen retrieveNino andThen retrieveBtaNavBar).async {
       implicit user =>
-        calculationService.getLiabilityCalculationDetail(user.mtditid, user.nino, taxYear).map {
-          case liabilityCalc: LiabilityCalculationResponse =>
-            val viewModel = AllowancesAndDeductionsViewModel(liabilityCalc.calculation)
-            auditingService.extendedAudit(AllowanceAndDeductionsResponseAuditModel(user, viewModel))
-            Ok(deductionBreakdownView(viewModel, taxYear, backUrl(taxYear), btaNavPartial = user.btaNavPartial))
-          case error: LiabilityCalculationError if error.status == NOT_FOUND =>
-            Logger("application").info(s"[DeductionsSummaryController][showDeductionsSummary[$taxYear]] No deductions data found.")
-            itvcErrorHandler.showInternalServerError()
-          case _: LiabilityCalculationError =>
-            Logger("application").error(
-              s"[DeductionsSummaryController][showDeductionsSummary[$taxYear]] No new calc deductions data error found. Downstream error")
-            itvcErrorHandler.showInternalServerError()
-        }
+        handleRequest(
+          backUrl = controllers.routes.TaxYearOverviewController.renderTaxYearOverviewPage(taxYear).url,
+          itcvErrorHandler = itvcErrorHandler,
+          taxYear = taxYear,
+          isAgent = false
+        )
     }
-  }
 
-  def backUrl(taxYear: Int): String = controllers.routes.TaxYearOverviewController.renderTaxYearOverviewPage(taxYear).url
-}
+  def showDeductionsSummaryAgent(taxYear: Int): Action[AnyContent] = {
+    Authenticated.async { implicit request =>
+      implicit agent =>
+        handleRequest(
+          backUrl = controllers.agent.routes.TaxYearOverviewController.show(taxYear).url,
+          itcvErrorHandler = itvcErrorHandlerAgent,
+          taxYear = taxYear,
+          isAgent = true
+        )(getMtdItUserWithNino()(agent, request, implicitly), implicitly, implicitly, implicitly)
+    }
+  }
+}
```

Comment:
newline pls :)

Reply:
Done, thanks. But intelij Idea Default Scala style formatting is not adding extra line here. And I thought we've agreed to use Ctrl + Alt + L for the default format